extensions/wikia/Search/classes/Services/ESFandomSearchService.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+class ESFandomSearchService extends AbstractSearchService {
+	const RESULTS_COUNT = 6;
+
+	protected function getConsulUrl() {
+		global $wgConsulServiceTag, $wgConsulUrl;
+
+		return ( new \Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl,
+				$wgConsulServiceTag ) )->getUrl( 'fandom-search' );
+	}
+
+	protected function prepareQuery( string $query ) {
+		return urlencode( $query );
+	}
+
+	protected function select( $query ) {
+
+		$consulUrl = $this->getConsulUrl();
+
+		$response = \Http::post( "http://$consulUrl/fandom", array(
+			'noProxy' => true,
+			'postData' => "queryTerms=$query",
+			'headers' => [
+				'Content-Type' => 'application/x-www-form-urlencoded',
+			],
+		) );
+		if ( $response !== false ) {
+			$decodedResponse = json_decode( $response, true );
+			if ( json_last_error() !== JSON_ERROR_NONE ) {
+				\WikiaLogger::instance()->error( " Fandom Stories Search: error decoding response", [
+					'query' => $query,
+					'response' => $response,
+					'error' => json_last_error(),
+				] );
+			} else if ( isset( $decodedResponse['hits']['hits'] ) ) {
+				return $decodedResponse['hits']['hits'];
+			} else {
+				\WikiaLogger::instance()->error( " Fandom Stories Search: invalid response", [
+					'query' => $query,
+					'response' => $response
+				] );
+			}
+		} else {
+			\WikiaLogger::instance()->error( " Fandom Stories Search: empty response", [
+				'query' => $query,
+			] );
+		}
+
+		// Must return an array of results or an empty array - don't return any errors here
+		return [];
+	}
+
+	protected function consumeResponse( $response ) {
+		$results = [];
+
+		foreach ( $response as $item ) {
+			if ( isset( $item['_source'] ) ) {
+				$source = $item['_source'];
+
+				$results[] = [
+					'title' => html_entity_decode( $source['title'] ),
+					'excerpt' => isset( $source['excerpt'] )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "excerpt" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)